pyccel/parser/scope.py

```diff
@@ -25,8 +29,10 @@ class Scope(object):
                  A dictionary of any decorators which operate on
                  objects in this scope
     """
-    __slots__ = ('_imports','_locals','parent_scope','_sons_scopes',
-            '_is_loop','_loops','_temporary_variables')
+    allow_loop_scoping = False
```

Comment:
`allow_loop_scoping` indicates whether a variable can be defined inside a scope. This is linked to the language. Currently `allow_loop_scoping` is always `False`. In python and Fortran it should remain `False` as these languages don't allow us to scope a variable to a loop. In the future it may be set to `True` once we reach `CCodePrinter` to allow temporaries created in loops to remain in these loops. As this concept is linked to the language and not any individual scope, I think it makes more sense to set it for all instances of the class rather than for individual scopes.